packages/host/tests/integration/store-test.gts

```diff
@@ -776,6 +778,24 @@ module('Integration | Store', function (hooks) {
     (instance as any).name = 'Air';
   });
 
+  test<TestContextWithSave>('an instance will NOT auto save when its data changes, if the user does not have write permissions', async function (assert) {
+    await realmService.logout(); //logout means store does not have write access to the realm
```

Comment:
this seems like an awkward way of saying this. it's not inconceivable that there might be other side effects of logging out that would impact the store. can we just mock the realm service response that would indicate that this is read only so we hold everything else about the realm service constant.

Reply:
Like just set our `RealmService.permissions()` to always return `{ canRead: true, canWrite: false}`.